### PR TITLE
Add support to list type

### PR DESCRIPTION
**Pull request by yim-lee**
- Add support for read/write collection `list` column
- Add support for querying with parameter of array type (e.g., `select ... in <array>`)

Currently `[Int8]`, `[Int16]`, `[Int32]`, `[Int64]`, `[Float32]`, `[Double]`, and `[String]` are supported.

Resolves https://github.com/apple/swift-cassandra-client/issues/24

### CODE REVIEW

File: Sources/CassandraClient/Data.swift [hunk 1]
```diff
@@ -593,11 +595,168 @@ extension CassandraClient.Column {
...
+
+        let iterator = cass_iterator_from_collection(self.rawPointer)
+        while cass_iterator_next(iterator) == cass_true {
+            let valuePointer = cass_iterator_get_value(iterator)
+            let value: T?
+            switch type {
+            case is Int8.Type:
+                var v: Int8 = 0
+                let error = cass_value_get_int8(valuePointer, &v)
+                value = error == CASS_OK ? v as! T? : nil
```
Comment (tomerd) [inline Sources/CassandraClient/Data.swift:656]: `as?`

**Review by tomerd — APPROVED** [review]

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Pull request merged**